HLTrigger/HLTcore/interface/HLTPrescaleProvider.h

```diff
@@ -138,8 +138,8 @@ template <>
 FractionalPrescale HLTPrescaleProvider::convertL1PS<FractionalPrescale>(double val) const;
 
 template <>
-unsigned int HLTPrescaleProvider::prescaleValue<unsigned int>(const edm::Event& iEvent,
-                                                              const edm::EventSetup& iSetup,
-                                                              const std::string& trigger);
+double HLTPrescaleProvider::prescaleValue<double>(const edm::Event& iEvent,
+                                                  const edm::EventSetup& iSetup,
+                                                  const std::string& trigger);
```

Comment:
I dont understand. Wouldnt this return only integer prescale values?

Reply:
This is a template specialisation for unsigned int (which returns 1 instead of -1 in case a prescale is not found, because unsigned int can not use a value of -1).

This is independent from the fact that with fractional prescales, we should remove return types of int and unsigned int a allowed types, see also https://github.com/cms-sw/cmssw/blob/master/HLTrigger/HLTcore/interface/HLTConfigProvider.h#L186-L195

So I suggest to revert from double back to unsigned int for the template specialisation (as 1 instead of -1 is ONLY needed for unsigned int), and later remove possibilities of using int or unsigned int alltogether, once client code has been fixed.